targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs

```diff
@@ -0,0 +1,205 @@
+using System.Collections;
+using System.IO;
+using PlayFab.ClientModels;
+using PlayFab.Json;
+using PlayFab.SharedModels;
+using UnityEngine;
+
+
+#if !UNITY_WSA && !UNITY_WP8
+using Ionic.Zlib;
+#endif
+
+namespace PlayFab.Internal
+{
+#if UNITY_2017_1_OR_NEWER
+
+    using UnityEngine.Networking;
+
+    public class PlayFabUnityHttp : IPlayFabHttp
+    {
+        private readonly int _pendingWwwMessages = 0;
+
+        public bool SessionStarted { get; set; }
+        public string AuthKey { get; set; }
+        public string EntityToken { get; set; }
+
+        public void InitializeHttp() { }
+
+        public void Update() { }
+
+        public void OnDestroy() { }
+
+        public void MakeApiCall(CallRequestContainer reqContainer)
+        {
+            reqContainer.RequestHeaders["Content-Type"] = "application/json";
+
+#if !UNITY_WSA && !UNITY_WP8 && !UNITY_WEBGL
+            if (PlayFabSettings.CompressApiData)
+            {
+                reqContainer.RequestHeaders["Content-Encoding"] = "GZIP";
+                reqContainer.RequestHeaders["Accept-Encoding"] = "GZIP";
+
+                using (var stream = new MemoryStream())
+                {
+                    using (var zipstream = new GZipStream(stream, CompressionMode.Compress,
+                        CompressionLevel.BestCompression))
+                    {
+                        zipstream.Write(reqContainer.Payload, 0, reqContainer.Payload.Length);
+                    }
+                    reqContainer.Payload = stream.ToArray();
+                }
+            }
+#endif
+
+            // Start the www corouting to Post, and get a response or error which is then passed to the callbacks.
+            PlayFabHttp.instance.StartCoroutine(Post(reqContainer));
+        }
+
+        private IEnumerator Post(CallRequestContainer reqContainer)
+        {
+#if PLAYFAB_REQUEST_TIMING
+                var stopwatch = System.Diagnostics.Stopwatch.StartNew();
+            #endif
+
+            var www = new UnityWebRequest(reqContainer.FullUrl)
+            {
+                uploadHandler = new UploadHandlerRaw(reqContainer.Payload),
+                downloadHandler = new DownloadHandlerBuffer(),
+                method = "POST"
+            };
+
+            foreach (var headerPair in reqContainer.RequestHeaders)
+                www.SetRequestHeader(headerPair.Key, headerPair.Value);
+
+            yield return www.SendWebRequest();
+
+#if PLAYFAB_REQUEST_TIMING
+            stopwatch.Stop();
+            var timing = new PlayFabHttp.RequestTiming {
+                StartTimeUtc = startTime,
+                ApiEndpoint = reqContainer.ApiEndpoint,
+                WorkerRequestMs = (int)stopwatch.ElapsedMilliseconds,
+                MainThreadRequestMs = (int)stopwatch.ElapsedMilliseconds
+            };
+            PlayFabHttp.SendRequestTiming(timing);
+#endif
+
+            if (!string.IsNullOrEmpty(www.error))
+                OnError(www.error, reqContainer);
```

Comment:
Bracket inconsistency. If there are brackets on the if/else, then the whole if-switch should have brackets.  Don't mix.